storage/src/Database.cpp

```diff
@@ -19,10 +20,12 @@ Database::~Database() {
     std::cerr << "~Database\n";
 }
 
-Database::Database(const std::string& db_path)
-    : cleanup_timer(new Timer(std::bind(&Database::perform_cleanup, this))) {
+Database::Database(boost::asio::io_context& ioc, const std::string& db_path)
+    : cleanup_timer_(ioc) {
     open_and_prepare(db_path);
-    cleanup_timer->start();
+
+    cleanup_timer_.expires_after(CLEANUP_PERIOD);
+    cleanup_timer_.async_wait(std::bind(&Database::perform_cleanup, this));
```

Comment:
How about running `perform_cleanup` at startup to minimise code duplication?

Reply:
At the time I thought it would be wasteful to clean an empty DB, but now I realise that this DB won't be new most of the time. Updated now.